io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/SerializerFactory.java

```diff
@@ -24,111 +24,102 @@
+        if (serializers.containsKey(serializer.inputClass())) {
+            if (avoidDuplication) {
+                throw new IllegalArgumentException("The Serializer for " + serializer.inputClass().getName() + " already exists!");
+            } else {
+                LOGGER.warn("Serializer with key type " + serializer.inputClass().getName() + " already exists, it is recommended to avoid duplication.");
+            }
+            return;
         }
 
-        serializerQueue = null;
-    }
+        SerializerData serializerData = new SerializerData(serializer, avoidDuplication);
 
-    @Nullable
-    public ObjectSerializer<?, ?> findSerializer(Class<?> type) {
-        return this.serializers.getOrDefault(type, null);
+        this.serializers.put(serializer.inputClass(), serializerData);
+        this.jacksonService.registerJacksonConfigure(new SerializerJacksonConfigure(serializer));
     }
 
-    public static class JacksonSerailizer extends StdSerializer {
-
-        private final ObjectSerializer serializer;
-
-        public JacksonSerailizer(ObjectSerializer serializer) {
-            super(serializer.inputClass());
-
-            this.serializer = serializer;
-        }
-
-        @Override
-        public void serialize(Object o, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
-           jsonGenerator.writeObject(serializer.serialize(o));
-        }
+    /**
+     * Unregister an existing serializer.
+     * @param serializer The serializer instance
+     * @return true if unregister success
+     */
+    public boolean unregisterSerializer(@NotNull ObjectSerializer<?, ?> serializer) {
+        return this.serializers.remove(serializer.inputClass()) != null;
     }
 
-    public static class JacksonDeserailizer extends StdDeserializer {
-
-        private final ObjectSerializer serializer;
-
-        public JacksonDeserailizer(ObjectSerializer serializer) {
-            super(serializer.inputClass());
-
-            this.serializer = serializer;
-        }
-
+    /**
+     * Unregister an existing serializer.
+     * @param type The serializer key type
+     * @return true if unregister success
+     */
+    public boolean unregisterSerializer(@NotNull Class<?> type) {
+        return this.serializers.remove(type) != null;
+    }
 
-        @Override
-        public Object deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
-            return serializer.deserialize(jsonParser.readValueAs(serializer.outputClass()));
-        }
+    /**
+     * Search for the serializer instance by key type.
+     * @param type the type of the serializer you are looking for
+     * @return the serializer instance, null if not found
+     */
+    @Nullable
+    public ObjectSerializer<?, ?> findSerializer(@NotNull Class<?> type) {
+        final SerializerData serializerData = this.serializers.getOrDefault(type, null);
```

Comment:
Doing the get on a concurrent hashmap also returns null if the key is absent.

Reply:
Fixed